src/ui/common/src/components/workflows/artifact/metric/history.tsx

```diff
@@ -49,16 +49,24 @@ const MetricsHistory: React.FC<Props> = ({ historyWithLoadingStatus }) => {
     schema: metricHistorySchema,
     data: (historyWithLoadingStatus.results?.results ?? []).map(
       (artifactStatusResult) => {
+        let timestamp = new Date(
+          artifactStatusResult.exec_state?.timestamps?.finished_at
+        ).toLocaleString();
+
+        // Metrics that are canceled / fail to execute have no exec_state, and thus no date.
+        if (timestamp === 'Invalid Date') {
+          timestamp = 'Unknown';
```

Comment:
Same comment as in check history

Reply:
@eunice-chan See render here as well, line 152, status gets shown as a StatusIndicator component:
```
                <StatusIndicator status={entry.status as ExecutionStatus} />

                <Typography sx={{ ml: 1 }} variant="body2">
                  {entry.timestamp}
                </Typography>
              </Box>
```